### PR TITLE
Unified base mode colors and helper function

### PR DESCRIPTION
**Pull request by louisg1337**
Moved `modeColors`, `BaseModes`, and `getBaseModeByKey` from [`diaryHelper.ts`](https://github.com/e-mission/e-mission-phone/blob/master/www/js/diary/diaryHelper.ts) to `e-mission-common` for centralized use between `e-mission` platforms. 

**Testing Done**
- Used npm link to add my local `e-mission-common` as a dependency in `e-mission-phone`
- Then replaced the objects and function in favor of our `e-mission-common` version as shown below.
  ```
  export const modeColors = base_mode_colors.modeColors;
  export const BaseModes = base_mode_colors.BaseModes;
  export const getBaseModeByKey = base_mode_colors.getBaseModeByKey;
  ```

### CODE REVIEW

File: src/emcommon/diary/base_mode_colors.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+modeColors = {
```
Comment (JGreenlee) [inline src/emcommon/diary/base_mode_colors.py:1]:
```suggestion
mode_colors = {
```
Use snake_case for python code

File: src/emcommon/diary/base_mode_colors.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "orange": '#bf5900', # oklch(58% 0.16 50)     # air, hsr
+    "green": '#008148', # oklch(53% 0.14 155)    # bike, e-bike, moped
+    "blue": '#0074b7', # oklch(54% 0.14 245)    # walk
+    "periwinkle": '#6356bf', # oklch(52% 0.16 285)    # light rail, train, tram, subway
+    "magenta": '#9240a4', # oklch(52% 0.17 320)    # bus
+    "grey": '#555555', # oklch(45% 0 0)         # unprocessed / unknown
+    "taupe": '#7d585a', # oklch(50% 0.05 15)     # ferry, trolleybus, user-defined modes
+}
+
+BaseModes = {
```
Comment (JGreenlee) [inline src/emcommon/diary/base_mode_colors.py:13]:
```suggestion
BASE_MODES = {
```
uppercase SNAKE_CASE to signify that this is a constant

File: src/emcommon/diary/base_mode_colors.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  "TRAIN": { "name": 'TRAIN', "icon": 'train-car-passenger', "color": modeColors.periwinkle },
+  "TRAM": { "name": 'TRAM', "icon": 'fas fa-tram', "color": modeColors.periwinkle },
+  "SUBWAY": { "name": 'SUBWAY', "icon": 'subway-variant', "color": modeColors.periwinkle },
+  "FERRY": { "name": 'FERRY', "icon": 'ferry', "color": modeColors.taupe },
+  "TROLLEYBUS": { "name": 'TROLLEYBUS', "icon": 'bus-side', "color": modeColors.taupe },
+  "UNPROCESSED": { "name": 'UNPROCESSED', "icon": 'help', "color": modeColors.grey },
+  "OTHER": { "name": 'OTHER', "icon": 'pencil-circle', "color": modeColors.taupe },
+};
+
+def getBaseModeByKey(motionName):
```
Comment (JGreenlee) [inline src/emcommon/diary/base_mode_colors.py:40]:
```suggestion
def get_base_mode_by_key(motionName):
```
snake_case

File: src/emcommon/diary/base_mode_colors.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    key = ('' + motionName).toUpperCase()
+    pop = key.split('.').pop() # if "MotionTypes.WALKING", then just take "WALKING"
+    return (pop and BaseModes[pop]) or BaseModes["UNKNOWN"]
```
Comment (JGreenlee) [inline src/emcommon/diary/base_mode_colors.py:43]: `toUpperCase`, `split`, `pop` are Javascript methods that don't exist in python. the compiled JS might work, but this wouldn't work in Python.

rewrite with the appropriate python methods and then test the compiled JS

**Review by JGreenlee — CHANGES_REQUESTED** [review]
Great job setting this up. Glad you were able to use the compile script and use npm link to test the compiled JS in the context of e-mission-phone

You should try to test the source Python code as well

**Comment by louisg1337** [comment]
Thanks for the feedback @JGreenlee! I took some time to incorporate your requests and actually properly test the python implementation, in which I found a few more bugs that I also addressed in this latest push. Also, I did have to change `toUpperCase()` to `upper()`, but apparently `split()` and `pop()` are both functions in python as well, so I ended up getting lucky with that. 

**Testing Done**
- I did the same tests as above, where I plugged in our `e-mission-common`'s versions of our functions in `diaryHelper.ts` and I got no errors.
- I tested the python implementation of `get_base_mode_by_key()` as follows...
  ```py
  print(get_base_mode_by_key("AIR"))
  print(get_base_mode_by_key("HELLO"))
  > {'name': 'AIR', 'icon': 'airplane', 'color': '#bf5900'}
  > {'name': 'UNKNOWN', 'icon': 'help', 'color': '#555555'}
  ```
- I then did the same for the javascript implementation...
  ```js
  console.log(get_base_mode_by_key("AIR"))
  console.log(get_base_mode_by_key("HELLO"))
  > {'name': 'AIR', 'icon': 'airplane', 'color': '#bf5900'}
  > {'name': 'UNKNOWN', 'icon': 'help', 'color': '#555555'}
  ```

**Review by JGreenlee — APPROVED** [review]
Great! Now can you create a corresponding PR on e-mission-phone? Search for places where `getBaseModeByKey` was imported from `diaryHelper`. Replace those by importing `base_mode_colors` from `e-mission-common` and calling `base_mode_colors.get_base_mode_by_key`.
Then you can remove `getBaseModeByKey`.

**Comment by louisg1337** [comment]
Sounds good, just released a [PR](https://github.com/e-mission/e-mission-phone/pull/1164) with those changes!

**Pull request merged**